linode/manager
# fix: [M3-7356] - VPC IPv4 checkboxes in Linode Config dialog

## Description 📝
Fix VPC IPv4 checkbox and input display when editing a config that already has ipv4 data

## Changes  🔄
1. The "Assign a public IPv4 address" checkbox is now checked in the Edit Config dialog upon opening if a VPC interface has already been 1:1 NAT’ed (i.e., if the `ipv4.nat_1_1` field is populated)
2. The "Auto-assign a VPC IPv4 address" is now unchecked in the Edit Config dialog if the `ipv4.vpc` field is already populated and we just expose the VPC IPv4 input field (pre-filled with the `ipv4.vpc` value)

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115299789/39dbe0cc-5147-4b55-b499-b0f1ac55d205" /> | <video src="https://github.com/linode/manager/assets/115299789/b1e708d5-5628-4e1c-b35c-461f535c9070" /> |

(Linode was deleted after)

## How to test 🧪

### Prerequisites
- Point to the dev env and ensure your account has vpc customer tags
- Create a VPC and Subnet if you don't already have one
- Click on Add Configuration in the Linode details -> Configuration tab
  - Provide a label and scroll down to the Networking section
  - Add a VPC, Subnet, and check both of the checkboxes

### Reproduction steps
- Be on the develop branch or go to the remote dev environment
- Filter your Network requests by `config` and notice how the `ipv4.vpc` and `ipv4.nat_1_1` fields of the interface are populated
- Click on Edit Configuration. The "Auto-assign a VPC IPv4 address" checkbox is checked and the VPC IPv4 input field is not exposed despite the field being populated. And the "Assign a public IPv4 address" checkbox is unchecked

### Verification steps 
- Filter your Network requests by `config` and notice how the `ipv4.vpc` and `ipv4.nat_1_1` fields of the interface are populated
- Click on Edit Configuration. The "Auto-assign a VPC IPv4 address" checkbox should be unchecked and the VPC IPv4 input field is now exposed and populated with the `ipv4.vpc` value. And the "Assign a public IPv4 address" checkbox is now checked
 ```
yarn test VPCPanel
```

## As an Author I have considered 🤔

*Check all that apply*
